build-system/babel-plugins/babel-plugin-transform-jss/index.js

```diff
@@ -109,25 +117,23 @@ module.exports = function ({template, types: t}) {
           );
         }
 
-        // Create the classes var.
-        const id = path.scope.generateUidIdentifier('classes');
-        const init = t.valueToNode(sheet.classes);
-        path.scope.push({id, init});
-        path.scope.bindings[id.name].path.parentPath.addComment(
-          'leading',
-          '* @enum {string}'
-        );
-
-        // Replace useStyles with a getter for the new `classes` var.
-        path.replaceWith(template.expression.ast`(() => ${t.cloneNode(id)})`);
-
         // Export a variable named CSS with the compiled CSS.
         const {css} = transformCssSync(sheet.toString());
         const cssStr = t.stringLiteral(css);
         const cssExport = template.ast`export const CSS = ${cssStr}`;
         path
           .findParent((p) => p.type === 'ExportNamedDeclaration')
           .insertAfter(cssExport);
+
+        path.replaceWith(template.ast`(() => ({}))`);
+
+        const {jssMappingsFile} = state.opts;
+        if (jssMappingsFile) {
+          writeJsonSync(jssMappingsFile, {
```

Comment:
I'm not sure this json indirection is really necessary. Why not just export multiple class names from the file?

```js
// component.jss.js

// input
export const useStyles = createUseStyles({
  foo: 'bar',
});

// output
export const useStyles = () => ({});
export const useStylesFoo = 'bar';
```

```js
// component.js

// input
import { useStyles } from './component.jss.js';
useStyles().foo;

// output
import { useStylesFoo } from './component.jss.js';
useStylesFoo;
```

Reply:
You need the indirection because you need to know the names like `useStylesFoo`, unless we make those deterministic and risk colliding scope values.

Reply:
Actually, you're right. It might be best to use deterministic names, avoiding the file altogether. It can also be done in one transform that way. 

To prevent name collision `$` is used (since it doesn't tend to appear normally), and local imports are decorated with their `useStyles` import name (some components may import more than one JSS file).

```js
// component.jss.js

// input
export const useStyles = createUseStyles({
  foo: 'bar',
});

// output
export const useStyles = () => ({});
export const $foo = 'bar';
```

```js
// component.js

// input
import { useStyles as myUseStyles } from './component.jss.js';
useStyles().foo;

// output
import { $foo as myUseStyles_$foo } from './component.jss.js';
myUseStyles_$foo;
```